yt_dlp/YoutubeDL.py

```diff
@@ -713,67 +714,33 @@ def preload_download_archive(fn):
 
-    @staticmethod
-    def make_logger(params):
-        screen = sys.stderr if params.get('logtostderr') else sys.stdout
-        stdout = sys.stderr if params.get('quiet') else screen
-        verbosity = (
-            Verbosity.VERBOSE if params.get('verbose')
-            else Verbosity.QUIET if params.get('quiet')
-            else Verbosity.NORMAL)
-
-        logger = Logger(
-            screen, verbosity, encoding=params.get('encoding'),
-            allow_color=not params.get('nocolor', False),
-            disable_progress=bool(params.get('noprogress')))
-
-        logger_param = params.get('logger')
-        if logger_param:
-            if logger_param == 'logging':
-                logger.setup_logging_logger()
-            else:
-                logger.setup_class_logger(logger_param)
+    def _setup_output(self):
+        logger = self.params.get('logger')
+        if isinstance(logger, Logger):
+            logger = output_helper.wrap_debug(logger)
         else:
-            logger.setup_stream_logger(stdout, sys.stderr, no_warnings=params.get('no_warnings', False))
-
-        return logger
+            logger = output_helper._make_ydl_logger(self.params)
 
-    def _setup_output(self, params):
-        def debug(func):
-            def wrapper(message, once=True):
-                func(f'[debug] {message}', once=once)
-
-            return wrapper
+        ignore_errors = bool(self.params.get('ignoreerrors'))
 
         def handle_error(func):
-            def wrapper(message, trace=None, is_error=True, prefix=True):
-                func(message, trace=trace, is_error=is_error, prefix=prefix)
+            def wrapper(message, tb=None, is_error=True, prefix=True):
+                func(message, tb=tb, is_error=is_error, prefix=prefix)
                 if not is_error:
                     return
-                if not self.params.get('ignoreerrors'):
+                if not ignore_errors:
                     raise DownloadError(message, sys.exc_info())
 
                 self._download_retcode = 1
 
             return wrapper
 
-        windows_enable_vt_mode()
-
-        self.logger = self.make_logger(params).make_derived(debug=debug, handle_error=handle_error)
-        if params.get('bidi_workaround', False):
-            try:
-                self.logger.init_bidi_workaround()
-
-            except OSError as ose:
-                if ose.errno != errno.ENOENT:
-                    raise
-
-                self.logger.warning(
-                    'Could not find any bidi executable, ignoring --bidi-workaround. '
-                    'Make sure that either bidiv or fribidi are available as executables in your $PATH.')
+        _error_wrap_indicator = '__ydl_error_wrapped'
+        if not getattr(logger, _error_wrap_indicator, None):
+            logger = logger.derive(handle_error=handle_error)
+            setattr(logger, _error_wrap_indicator, True)
```

Comment:
This causes problems with double derivation
```py
a = YoutubeDL()
b = YoutubeDL(dict(logger=a.logger))
```
We need to unwrap a logger before using it (and a way to do that)